### PR TITLE
Curl scalars 4.2 [help needed - tests]

### PR DESCRIPTION
**Pull request by nowackipawel**
#4911

### CODE REVIEW

File: system/HTTP/CURLRequest.php [hunk 1]
```diff
@@ -309,7 +309,11 @@ protected function parseOptions(array $options)
 
         if (array_key_exists('headers', $options) && is_array($options['headers'])) {
             foreach ($options['headers'] as $name => $value) {
-                $this->setHeader($name, $value);
+                if (is_scalar($value)) {
+                    $this->setHeader($name, $value);
```
Comment (MGatner) [inline system/HTTP/CURLRequest.php:313]: Is there any reason to include stringables?
```suggestion
                if (is_scalar($value) || (is_object($value) && method_exists($value, '__toString')) {
                    $this->setHeader($name, (string) $value);
```

**Review by MGatner — CHANGES_REQUESTED** [review]
Looking better! Still need tests.

**Comment by nowackipawel** [comment]
@MGatner in this case I would say we need to add label "help needed". :(.
I would love to write tests as I did  year or two ago, but at the moment hard to find time for that kinda things :(.

**Comment by lonnieezell** [comment]
> I would love to write tests as I did year or two ago, but at the moment hard to find time for that kinda things :(.

@nowackipawel  We all completely understand time constraints. We are all volunteers stealing  time from the rest of our life to work on this. These look like good changes. Is there no way you can take a little time to learn testing? There's a better chance of getting changes in (of which you've had quite a few good suggestions over the years) if you can do the testing. Otherwise you're asking us to take time from what we see as needs to test your code. And we all have limited time to work on this.

**Comment by kenjis** [comment]
It seems the base branch of this PR must be changed to `develop`.

**Comment by kenjis** [comment]
@nowackipawel 
There are conflicts.

Could you rebase your branch?
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#pushing-your-branch

**Comment by paulbalandan** [comment]
Hi, and thanks for your contribution! As this PR has been standing for a couple of months now and no effort has been seen to add the necessary unit testing to confirm the patch, we can conclude that there's no more interest in moving this forward.

We love to see and appreciate all contributions coming in from the community. However, please make it an effort to make it easy for maintainers to verify your code (that you want to be part of the framework core) through unit testing. You cannot (and we cannot, either) expect to write only source code and delegate the testing part to others. Unfortunately, that's not how OSS works.

Closing this for now. We'd still love to see you around contributing to the framework. Feel free to open a new PR once all necessary testing is in place. Thanks!

**Pull request closed without merging**